app/components/file/browse/blob-files-browser/blob-files-browser.component.ts

```diff
@@ -52,11 +57,36 @@ export class BlobFilesBrowserComponent implements OnChanges, OnDestroy {
                 this.upload(event).subscribe(() => {
                     this.fileNavigator.refresh(path);
                 });
+
                 return Observable.of(null);
             },
         });
     }
 
+    public handleDeleteEvent(event: FileDeleteEvent) {
+        const { path } = event;
+        this.dialogService.confirm(`Delete files`, {
+            description: event.isDirectory
+                ? `All files will be deleted from the folder: ${path}`
+                : `The file '${FileUrlUtils.getFileName(path)}' will be deleted.`,
+            yes: () => {
+                const listParams = { recursive: true, startswith: path };
+                const data = this.storageService.listBlobs(this.container, listParams);
+                const obs = data.fetchAll().flatMap(() => data.items.take(1));
```

Comment:
add .shareReplay(1) at the end so it doesn't trigger the flatMap for every subscription